### PR TITLE
Updating binaryen, wasm-install and enabling V8 tests on Mac

### PR DESCRIPTION
**Pull request by eholk**
This PR includes a number of updates that ended up all being tangled together.

* `build.rs` now uses [curl](https://crates.io/crates/curl) to fetch the appropriate wasm-install package.
* Since the waterfall at http://wasm-stat.us now includes Mac builds, this PR enables downloading wasm-install on Mac and running the test suite in V8.
* This PR updates the wasm-install versions to more recent builds. The new builds use Wasm binary format `0xd` instead of `0xc`, so this required...
* updating Binaryen to the latest master.
* Trans now adds an `unreachable` expression after calls to non-returning functions such as `panic`. This is needed to make sure the resulting Wasm bytes typecheck.

### CODE REVIEW

**Comment by eholk** [comment]
@dschuff - can you take a look at this too?

File: build.rs [hunk 1]
```diff
@@ -1,41 +1,67 @@
...
+    pub const DOWLOAD_WASM: bool = true;
+    pub const WASM_BUILD: &'static str = "2670";
+    pub const OS_NAME: &'static str = "mac";
+}
+
+#[cfg(not(any(target_os="linux", target_os="macos")))]
+mod config {
+    pub const DOWLOAD_WASM: bool = false;
+    pub const WASM_BUILD: &'static str = "";
+    pub const OS_NAME: &'static str = "";
```
Comment (dschuff) [inline build.rs:30]: FWIW the windows bot is now uploading a build as well. It doesn't run all of the tests yet but I think the binary is working. Obviously don't need to update this PR for that necessarily but FYI.

**Review by dschuff — COMMENTED** [review]
Looks OK to me.

File: build.rs (hunk 1 above)
Comment (eholk) [reply build.rs:30]: Awesome! I opened #55 so we don't forget to add this eventually.

**Pull request merged**